docs/release_guidelines.md

```diff
@@ -1,23 +1,115 @@
+
+N: Standard use-case of deployment and testing the version specified in the release branch name.
+N-1: For upgrade purposes. We need to be able to deploy N-1, run pre-upgrade tests, upgrade and reload configs for version N, then run post-upgrade tests. This process requires that release branches support the deployment and testing of previous versions.
+
+For example, the branch release-4.8 will be able to deploy OCP/ODF 4.8 as well as 4.7 to support upgrade testing. However, it will not officially support deployment or testing of versions <= 4.6.
+
+
+## Usage
+
+### Locally
+
+As these are simply git branches we are creating, you just need to perform a git checkout on the specific release branch you wish to run your deployment or tests with in order to use it.
+
+```
+git fetch upstream
+git checkout release-4.12
+```
+
+Note that `upstream` is the name of the remote for the base repository, not your personal fork.
+
+### Jenkins
+
+From Jenkins we will automatically be using the release branch if no other ocs-ci branch has been specified. Which specific release branch we use will be dynamically based on the `OCS_VERSION` or `UPGRADE_OCS_VERSION` parameters. For example, if `OCS_VERSION` is 4.13 we will use `release-4.13`. If this release branch doesn't exist yet, we will default back to the `stable` branch. This logic can be overridden by specifying a branch to use.
+
+## Creating Release Branches
+
+Release branches will branch off of the `stable` branch. Before we can create a release branch, we will need to know which commit to branch off from. Once we know the commit we are basing the branch off of, we can then go about creating the branch from either the git CLI or the Github UI.
+
+### Github UI
+
+From the Github UI, navigate to the list of [branches](https://github.com/red-hat-storage/ocs-ci/branches). From here you can click on new branch, selecting the `stable` branch as the base. Note that the branch creation process only works this way if the latest stable commit is the point we wish to branch off from. Otherwise we will need to create the branch using the CLI.
+
+### CLI
+
+First, ensure your local repository is up to date with the remote. Note that the following command is destructive so make sure you have any local changes stashed or committed before continuing.
+```
+git checkout master && git fetch --all && git reset --hard upstream/master
+```
+
+Checkout the stable branch
+```
+git checkout stable
+```
+
+Determine the starting point of the new release branch, for this you will need the git hash. You can find this using:
+```
+git log
+```
+
+Once you know the hash, we will create the new release branch using the hash from earlier as the base. Note that x.y is the major.minor version.
+```
+git checkout -b release-x.y 46568b9b4b8c2ceabad665b839655e0b19ab3634
+```
+
+Finally, push the branch to the remote.
+```
+git push upstream release-x.y
+```
```

Comment:
Also I would add info that those release branches are automatically protected and require the same PR validation checks as master branch, because of wildcard rule `release-*` we apply in the repository branches configuration.

Reply:
Added a note about this